src/utils/TextToSpeech.js

```diff
@@ -0,0 +1,74 @@
+const supportedVoiceLanguages = ['en', 'it'];
+let recommendedVoices = [];
+supportedVoiceLanguages.forEach(supportedVoiceLanguage => {
+  recommendedVoices[
+    supportedVoiceLanguage
+  ] = require(`./textToSpeech/${supportedVoiceLanguage}.json`);
+});
+
+export function hasTextToSpeechVoices() {
+  return speechSynthesis.getVoices().length !== 0;
+}
+
+export function startTextToSpeech(text, locale, onComplete) {
+  const voices = speechSynthesis.getVoices();
+  if (voices.length === 0) {
+    return false;
+  }
+
+  const filteredVoices = filterAvailableVoices(recommendedVoices[locale]);
+  if (filteredVoices.length === 0) {
+    return false;
+  }
+
+  const utterance = new SpeechSynthesisUtterance(text);
+
+  for (const voice of voices) {
```

Comment:
Potential follow-up: Should we store a list of `matchedVoicesByLocale` so we don't have to search the entire list every time?

Reply:
This does make sense longer term.  One reason I have it do the whole search each time now, rather than caching, is that it seems possible that the browser's voice list will refresh more than once as more voices are loaded; the documentation for the [SpeechSynthesis voiceschanged](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesis/voiceschanged_event) event isn't clear on whether this is likely to happen only once or more than once.  By reexamining the list of voices, it will give us the opportunity to hear if a better voice arrives partway through usage.